quickstart_template/jenkins-to-acr.sh

```diff
@@ -0,0 +1,113 @@
+  --registry_user_name|-ru              : Registry user name
+  --registry_password|-rp               : Registry password
+  --include_docker_build_pipeline|-i    : Include a docker build pipeline (off by default)
+EOF
+}
+
+function throw_if_empty() {
+  local name="$1"
+  local value="$2"
+  if [ -z "$value" ]; then
+    echo "Parameter '$name' cannot be empty." 1>&2
+    print_usage
+    exit -1
+  fi
+}
+
+while [[ $# > 0 ]]
+do
+  key="$1"
+  shift
+  case $key in
+    --vm_user_name|-u)
+      vm_user_name="$1"
+      shift
+      ;;
+    --git_url|-g)
+      git_url="$1"
+      shift
+      ;;
+    --registry|-r)
+      registry="$1"
+      shift
+      ;;
+    --registry_user_name|-ru)
+      registry_user_name="$1"
+      shift
+      ;;
+    --registry_password|-rp)
+      registry_password="$1"
+      shift
+      ;;
+    --include_docker_build_pipeline|-i)
+      include_docker_build_pipeline="$1"
+      shift
+      ;;
+    --help|-help|-h)
+      print_usage
+      exit 13
+      ;;
+    *)
+      echo "ERROR: Unknown argument '$key'" 1>&2
+      exit -1
+  esac
+done
+
+if [ -z "$include_docker_build_pipeline" ]
+then
+    include_docker_build_pipeline="0"
+fi
+
+if [[ "${include_docker_build_pipeline}" == "1" ]]
+then
+    throw_if_empty --vm_user_name $vm_user_name
+    throw_if_empty --git_url $git_url
+    throw_if_empty --registry $registry
+    throw_if_empty --registry_user_name $registry_user_name
+    throw_if_empty --registry_password $registry_password
+fi
+
+#install jenkins
+wget -q -O - https://pkg.jenkins.io/debian/jenkins-ci.org.key | sudo apt-key add -
+sudo sh -c 'echo deb http://pkg.jenkins.io/debian-stable binary/ > /etc/apt/sources.list.d/jenkins.list'
+sudo apt-get update --yes
+sudo apt-get install jenkins --yes
+sudo apt-get install jenkins --yes # sometime the first apt-get install jenkins command fails, so we try it twice
+sudo apt-get install git --yes
+sudo curl -sSL https://get.docker.com/ | sh
+sudo gpasswd -a jenkins docker
+sudo service docker restart
+skill -KILL -u jenkins
+sudo service jenkins restart
+
+if [[ "${include_docker_build_pipeline}" != "1" ]]
+then
+    echo "Skipping the pipeline inclusion"
+    exit 0
+fi
+
+echo "Including the pipeline"
+
+#download dependencies
+base_remote_jenkins_scripts="https://raw.githubusercontent.com/Azure/azure-devops-utils/master/jenkins"
+add_docker_build_job_script_name="add-docker-build-job.sh"
+
+curl "${base_remote_jenkins_scripts}/${add_docker_build_job_script_name}" -o ${add_docker_build_job_script_name}
+sudo chmod +x ${add_docker_build_job_script_name}
+
+#get password and call build creation script
+admin_password=`sudo cat /var/lib/jenkins/secrets/initialAdminPassword`
+
+./${add_docker_build_job_script_name} -j "http://localhost:8080/" -ju "admin" -jp "${admin_password}" -g "${git_url}" -r "${registry}" -ru "${registry_user_name}"  -rp "${registry_password}" -rr "${vm_user_name}/myfirstapp"
```

Comment:
You can run a script without saving it to disk like this:

`curl --silent "${base_remote_jenkins_scripts}/add-docker-build-job.sh" | sudo bash -s -- -j "http://localhost:8080/" -ju "admin" -jp "${admin_password}" -g "${git_url}" -r "${registry}" -ru "${registry_user_name}"  -rp "${registry_password}" -rr "${vm_user_name}/myfirstapp"`

NOTE: the '--' is important and I've missed it before. It separates the parameters for 'sudo bash' and the parameters for the downloaded script.